core/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -166,7 +187,7 @@ public int hashCode() {
     @Override
     public IndexTemplateMetaData readFrom(StreamInput in) throws IOException {
         Builder builder = new Builder(in.readString());
-        builder.order(in.readInt());
+        builder.order(in.readVInt());
```

Comment:
I think this can have an impact no? Even if we don't expect clusters with mixed alpha/beta/rc/ga versions of 5.0